src/test/scala/com/gd/twitteranalytics/TweetsTransformerTest.scala

```diff
@@ -0,0 +1,59 @@
+package twitteranalytics
+
+import java.sql.Timestamp
+import java.util.Date
+
+import com.gd.twitteranalytics.TweetsTransformer
+import com.holdenkarau.spark.testing.{DataFrameSuiteBase, StreamingSuiteBase}
+import org.scalatest.FreeSpec
+import twitter4j.{TwitterObjectFactory}
+import org.scalamock.scalatest.MockFactory
+
+import scala.io.Source
+
+class TweetsTransformerTest extends FreeSpec with StreamingSuiteBase with MockFactory  with DataFrameSuiteBase {
+
+  "TransformTweets" - {
+    val constDate = new Date("Wed Sep 26 10:37:20 PDT 2018")
+    val text = "This is the #Sample #text for #test purpose"
+    val origDate = new Timestamp(constDate.getTime)
+
+    "getDateAndText function" - {
+    "when presented with a complete tweet status" - {
+      "should retrieve date and text from the tweets status" in {
```

Comment:
all information available in a tweet event/status should be included (contributors, retweeted, etc.)

Reply:
Do you have any set of information or everything that is there in a Tweet Status ?

Reply:
is `tweetStatus.json` a comprehensive example of what can come as a status payload?

Reply:
anyway, everything should be included. no need to parse every parameter though.
Tweet model may look like this: `Tweet(created_at: Timestamp, payload: String)`
where `payload` incapsulates  raw status payload.

Reply:
parsing of parameters besides an event-time, should be avoided to enable forward  compatibility.

Reply:
Okay, So in this case what do I assert true in test case. Since the createStatus Api gives the status in different format. I can't equalize Json and created status.
val tweetStatus = TwitterObjectFactory.createStatus(rawJson)
val input = List(List(tweetStatus))
val expected = List(List((creationTime,rawJson)))
This clearly won't work..
testOperation(input,TweetsTransformer.getCreationDateAndStatus,expected,ordered = false)

Reply:
> the createStatus Api gives the status in different format...

I suppose that the "different format" is a java object.
in this case, you should encode the object to a JSON representation  and assign it to a `Tweet#payload`.

> what do I assert true in test case.

you should implement a test case that checks if a given `Status` gets correctly mapped to the `Tweet` model.